### ISSUE TITLE
ListView/ScrollView - Scrolling jumps to the bottom regardless of items on M1 iOS Simulator

### ISSUE THREAD

**Issue by ollyde** [flutter/flutter#79641]
Flutter 2.0.3
iOS Simulator 14.4
M1 Macbook Pro

Whole app code below.

The view jumps to the bottom; provided a gif example.

Normal behavior on device. iOS 14.4
![correct](https://user-images.githubusercontent.com/6940726/113418956-c43be780-93c6-11eb-8777-8fbb47a99fcb.gif)

Incorrect. Simulator 14.4
![error](https://user-images.githubusercontent.com/6940726/113418977-cd2cb900-93c6-11eb-89d2-fdce0fe78a57.gif)

App Code (Very standard)

```
import 'package:flutter/material.dart';

void main() {
  runApp(MyApp());
}

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      home: MyHomePage(title: 'Flutter Demo Home Page'),
    );
  }
}

class MyHomePage extends StatefulWidget {
  MyHomePage({Key key, this.title}) : super(key: key);
  final String title;

  @override
  _MyHomePageState createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  @override
  Widget build(BuildContext context) {
    return ListView(
      children: [
        Container(color: Colors.red, height: 100),
        Container(color: Colors.blue, height: 100),
        Container(color: Colors.red, height: 100),
        Container(color: Colors.blue, height: 100),
        Container(color: Colors.red, height: 100),
        Container(color: Colors.blue, height: 100),
        Container(color: Colors.red, height: 100),
        Container(color: Colors.blue, height: 100),
        Container(color: Colors.red, height: 100),
        Container(color: Colors.blue, height: 100),
        Container(color: Colors.red, height: 100),
        Container(color: Colors.blue, height: 100),
        Container(color: Colors.red, height: 100),
        Container(color: Colors.blue, height: 100),
        Container(color: Colors.red, height: 100),
        Container(color: Colors.blue, height: 100),
      ],
    );
  }
}
```

**Comment by iapicca** [comment]
@ollydixon 
I cannot test it myself (I'm not a mac user)
does it happens with builder as well?

```dart
void main() => runApp(
      MaterialApp(
        home: Material(
          child: ListView.builder(
            itemBuilder: (context, index) => index % 2 == 0
                ? Container(color: Colors.blue, height: 100)
                : Container(color: Colors.red, height: 100),
          ),
        ),
      ),
    );

```

**Comment by ollydixon** [comment]
@iapicca it happens on all types of scrolling views. Even SingleChild with a Colum inside of it.

**Comment by iapicca** [comment]
> it happens on all types of scrolling views. Even SingleChild with a Colum inside of it.
> 
> I tested Builder and SingleChild too. (I was trying to fix a bug for hours)

 @ollydixon 
ballistic looks super weird on `web` as well 
normal behavior on physical android device
 
 tested with this code https://github.com/flutter/flutter/issues/79641#issuecomment-812550952
 
 ```console
 [✓] Flutter (Channel master, 2.1.0-13.0.pre.416, on Linux, locale en_US.UTF-8)
    • Flutter version 2.1.0-13.0.pre.416 at
      /home/francesco/snap/flutter/common/flutter
    • Framework revision b9638fb837 (10 hours ago), 2021-04-02 00:39:02 -0400
    • Engine revision bed3a4a7a0
    • Dart version 2.13.0 (build 2.13.0-192.0.dev)

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
    • Android SDK at /home/francesco/Android/Sdk
    • Platform android-30, build-tools 30.0.2
    • ANDROID_SDK_ROOT = /home/francesco/Android/Sdk
    • Java binary at: /usr/lib/jvm/java-1.8.0-openjdk-amd64/jre/bin/java
    • Java version OpenJDK Runtime Environment (build
      1.8.0_282-8u282-b08-0ubuntu1~20.04-b08)
    • All Android licenses accepted.

[✓] Chrome - develop for the web
    • CHROME_EXECUTABLE = /snap/bin/chromium

[✓] Linux toolchain - develop for Linux desktop
    • clang version 6.0.0-1ubuntu2 (tags/RELEASE_600/final)
    • cmake version 3.10.2
    • ninja version 1.8.2
    • pkg-config version 0.29.1

[!] Android Studio (not installed)
    • Android Studio not found; download from
      https://developer.android.com/studio/index.html
      (or visit https://flutter.dev/docs/get-started/install/linux#android-setup
      for detailed instructions).

[✓] Connected device (3 available)
    • Pixel 3a (mobile) • 965AY0WP5C • android-arm64  • Android 11 (API 30)
    • Linux (desktop)   • linux      • linux-x64      • Linux
    • Chrome (web)      • chrome     • web-javascript • Chromium 89.0.4389.114
      snap

! Doctor found issues in 1 category.

 ```

**Comment by ollydixon** [comment]
@iapicca it's an insanly bad bug; are they even doing QA at Flutter Google? 😂

**Comment by TahaTesser** [comment]
Hi @ollydixon 
Thanks for catching this. It looks like it only reproduces when running on the simulator on M1. I have tried the same code sample on simulator running intel macOS as well as on a physical device, it doesn't reproduce anywhere but the M1 simulator. 

| M1 Simulator | Intel Simulator |
| --------------- | --------------- |
<img src="https://user-images.githubusercontent.com/48603081/113587594-acb76580-9637-11eb-826c-a0451d909994.gif" height="450" /> | <img src="https://user-images.githubusercontent.com/48603081/113587722-e12b2180-9637-11eb-9270-5d2a32d8762f.gif" height="450" />

 



<details> 
<summary>flutter doctor -v</summary> 

```bash
[✓] Flutter (Channel stable, 2.0.4, on macOS 11.2.3 20D91 darwin-arm, locale
    en-GB)
    • Flutter version 2.0.4 at /Users/nevercode/Code/flutter_stable
    • Framework revision b1395592de (4 days ago), 2021-04-01 14:25:01 -0700
    • Engine revision 2dce47073a
    • Dart version 2.12.2

[!] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at /Users/nevercode/Library/Android/sdk
    • Platform android-30, build-tools 30.0.3
    • Java binary at: /Users/nevercode/Library/Application
      Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/201.7042882/Android
      Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6915495)
    ✗ Android license status unknown.
      Run `flutter doctor --android-licenses` to accept the SDK licenses.
      See https://flutter.dev/docs/get-started/install/macos#android-setup for
      more details.

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.4, Build version 12D4e
    • CocoaPods version 1.10.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.1)
    • Android Studio at /Users/nevercode/Library/Application
      Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/201.7042882/Android
      Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6915495)

[✓] VS Code (version 1.55.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.21.0

[✓] Connected device (4 available)
    • Taha’s iPhone (mobile) • 00008020-001059882212002E            • ios
      • iOS 14.4.2
    • iPhone 11 (mobile)     • 7099B239-8266-4F66-9901-A229EFE0F43D • ios
      • com.apple.CoreSimulator.SimRuntime.iOS-14-4 (simulator)
    • macOS (desktop)        • macos                                •
      darwin-arm64   • macOS 11.2.3 20D91 darwin-arm
    • Chrome (web)           • chrome                               •
      web-javascript • Google Chrome 89.0.4389.114

! Doctor found issues in 1 category.
```

```bash
[✓] Flutter (Channel master, 2.1.0-13.0.pre.444, on macOS 11.2.3 20D91
    darwin-arm, locale en-GB)
    • Flutter version 2.1.0-13.0.pre.444 at /Users/nevercode/Code/flutter_master
    • Framework revision 824b1ad3f8 (2 days ago), 2021-04-03 15:24:02 -0400
    • Engine revision e393800c63
    • Dart version 2.13.0 (build 2.13.0-196.0.dev)

[!] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at /Users/nevercode/Library/Android/sdk
    • Platform android-30, build-tools 30.0.3
    • Java binary at: /Users/nevercode/Library/Application
      Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/201.7042882/Android
      Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6915495)
    ✗ Android license status unknown.
      Run `flutter doctor --android-licenses` to accept the SDK licenses.
      See https://flutter.dev/docs/get-started/install/macos#android-setup for
      more details.

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.4, Build version 12D4e
    • CocoaPods version 1.10.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.1)
    • Android Studio at /Users/nevercode/Library/Application
      Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/201.7042882/Android
      Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6915495)

[✓] VS Code (version 1.55.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.21.0

[✓] Connected device (4 available)
    • Taha’s iPhone (mobile) • 00008020-001059882212002E            • ios
      • iOS 14.4.2
    • iPhone 11 (mobile)     • 7099B239-8266-4F66-9901-A229EFE0F43D • ios
      • com.apple.CoreSimulator.SimRuntime.iOS-14-4 (simulator)
    • macOS (desktop)        • macos                                •
      darwin-arm64   • macOS 11.2.3 20D91 darwin-arm
    • Chrome (web)           • chrome                               •
      web-javascript • Google Chrome 89.0.4389.114

! Doctor found issues in 1 category.
```
	
</details>

**Comment by ollydixon** [comment]
@TahaTesser yes indeed; I also have a Intel MacOS build server and I was going insane trying to figure out what was causing this bug 😂🙈

**Comment by iapicca** [comment]
@TahaTesser 
do you experience "weird ballistics" on web as well?
https://github.com/flutter/flutter/issues/79641#issuecomment-812556405

**Comment by TahaTesser** [comment]
@iapicca 
I don't see any issue on the web using the code sample, if you're experiencing it please file a new issue for the web. Thanks!

**Comment by anmarb** [comment]
@TahaTesser 
The same issue can be experienced with Flutter 1.22.6 (and iOS Simulator 14.4, M1-based Mac). Any chance to also fix it for those projects not yet able to migrate?

**Comment by definitelyokay** [comment]
Also replicating on Flutter 1.22.5 (iOS Simulator 14.4, M1 Mac Mini). Can't migrate this project to 2.0 unfortunately, so fixing this on 1.22.x would be awesome.

![possessed_scrolling_2](https://user-images.githubusercontent.com/16472250/115452686-5a07ad00-a1e4-11eb-8621-ab83529249c1.gif)

**Comment by definitelyokay** [comment]
Forgot to mention that you can "work around" this by clicking very quickly (without moving the mouse) while it is bouncing toward the top or bottom.

This immediately stops the listview from scrolling any further and allows you to click whatever you needed to click. It took me a hot minute to figure that out so I figure I should share this.

Of course, this "work around" is only as good as your reflexes.

**Comment by iapicca** [comment]
> Also replicating on Flutter 1.22.5 (iOS Simulator 14.4, M1 Mac Mini). Can't migrate this project to 2.0 unfortunately, 
> ### so fixing this on 1.22.x would be awesome.

@definitelyokay fixes are rolled out on versions newer than the current stable... so AFAIK it won't happen

**Comment by sooxt98** [comment]
Ya the scrolling speed on M1 simulator is broken; one small touch will scroll to the end of the list

**Comment by jmagman** [comment]
I wanted to validate this was fixed by https://github.com/flutter/flutter/pull/85059 but I can't even reproduce it on Flutter 2.0.3.  Maybe Xcode fixed the bug in 12.5?  Seems unlikely though.


https://user-images.githubusercontent.com/682784/123025909-3a316880-d390-11eb-937f-368b335c374c.mov



```
$ flutter doctor -v
[✓] Flutter (Channel unknown, 2.0.3, on macOS 11.4 20F71 darwin-arm, locale en-US)
    • Flutter version 2.0.3 at /Users/flutter/Projects/flutter
    • Framework revision 4d7946a68d (3 months ago), 2021-03-18 17:24:33 -0700
    • Engine revision 3459eb2436
    • Dart version 2.12.2

[!] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at /Users/flutter/Library/Android/sdk
    • Platform android-30, build-tools 30.0.3
    • ANDROID_SDK_ROOT = /Users/flutter/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.8+10-b944.6916264)
    ✗ Android license status unknown.
      Run `flutter doctor --android-licenses` to accept the SDK licenses.
      See https://flutter.dev/docs/get-started/install/macos#android-setup for more details.

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.5.1, Build version 12E507
    • CocoaPods version 1.10.0

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.8+10-b944.6916264)

[✓] Connected device (3 available)
    • iPhone 11 (mobile) • 46A4D5A8-663C-49C2-9C1B-B97F3851FBD1 • ios            • com.apple.CoreSimulator.SimRuntime.iOS-14-5 (simulator)
    • macOS (desktop)    • macos                                • darwin-arm64   • macOS 11.4 20F71 darwin-arm
    • Chrome (web)       • chrome                               • web-javascript • Google Chrome 91.0.4472.114
```

**Comment by stenlee** [comment]
I can confirm the "broken scrolling" on M1 iOS Simulator is still available with latest Flutter 2.2.2
 Flutter (Channel stable, 2.2.2, on macOS 11.4 20F71 darwin-arm, locale en-CZ)

- the problem is also easily reproducible when using the Google Maps plugin https://pub.dev/packages/google_maps_flutter - moving with Map is almost impossible due to bad calculation of gestures)

the only solution by now is to run Simulator with Rosetta = with quite huge impact on performance :/

**Comment by jmagman** [comment]
Launching the simulator as a Rosetta app doesn't reproduce it?  That's really weird, the Flutter app itself is always running as a Rosetta app no matter how you launch the simulator (you can see the "Runner" process is "Intel" not "Apple" in Activity Monitor).

**Comment by jmagman** [comment]
That was indeed why I couldn't reproduce it--I had been running Simulator as a Rosetta app.  This is definitely a Rosetta Flutter app running on a non-Rosetta simulator bug.  We've seen other Rosetta simulator bugs like https://github.com/flutter/flutter/issues/74970#issuecomment-858170914.
Verified this will be fixed by #85059 (Flutter apps will run natively without Rosetta).

**Comment by admosity** [comment]
Thank you for reopening this issue! This is more so a developer experience annoyance, but would love to see this fixed ✨.


https://github.com/user-attachments/assets/7c9b245f-ef56-42b1-bb22-a69816d3d5a5



Flutter doctor:

```
[!] Flutter (Channel stable, 3.22.3, on macOS 14.5 23F79 darwin-arm64, locale en-US)
    • Flutter version 3.22.3 on channel stable at /Users/REDACTED/fvm/versions/3.22.3
    ! Warning: `flutter` on your path resolves to /Users/REDACTED/fvm/versions/3.22.2/bin/flutter, which is not inside your current Flutter SDK checkout at /Users/REDACTED/fvm/versions/3.22.3. Consider adding /Users/REDACTED/fvm/versions/3.22.3/bin to the front of your path.
    ! Warning: `dart` on your path resolves to /Users/REDACTED/fvm/versions/3.22.2/bin/dart, which is not inside your current Flutter SDK checkout at /Users/REDACTED/fvm/versions/3.22.3. Consider adding /Users/REDACTED/fvm/versions/3.22.3/bin to the front of your path.
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision b0850beeb2 (7 days ago), 2024-07-16 21:43:41 -0700
    • Engine revision 235db911ba
    • Dart version 3.4.4
    • DevTools version 2.34.3
    • If those were intentional, you can disregard the above warnings; however it is recommended to use "git" directly to perform update checks and upgrades.

[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.2)
    • Android SDK at /Users/REDACTED/Library/Android/sdk
    • Platform android-33, build-tools 33.0.2
    • Java binary at: /Applications/Android Studio.app/Contents/jbr/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.15+0-b2043.56-8887301)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 15.4)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 15F31d
    • CocoaPods version 1.15.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2022.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.15+0-b2043.56-8887301)

[✓] VS Code (version 1.91.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.92.0

[✓] Connected device (4 available)
    • iPhone 14 Pro Max (mobile)      • C3C647B8-8F34-47FE-AAF3-62F431F97F91 • ios            • com.apple.CoreSimulator.SimRuntime.iOS-17-4 (simulator)
    • macOS (desktop)                 • macos                                • darwin-arm64   • macOS 14.5 23F79 darwin-arm64
    • Mac Designed for iPad (desktop) • mac-designed-for-ipad                • darwin         • macOS 14.5 23F79 darwin-arm64
    • Chrome (web)                    • chrome                               • web-javascript • Google Chrome 126.0.6478.183

[✓] Network resources
    • All expected network resources are available.

! Doctor found issues in 1 category.
```

Device info:

```
Hardware Overview:

      Model Name: MacBook Air
      Model Identifier: Mac14,2
      Model Number: Z1610005FLL/A
      Chip: Apple M2
      Total Number of Cores: 8 (4 performance and 4 efficiency)
      Memory: 24 GB
      System Firmware Version: 10151.121.1
      OS Loader Version: 10151.121.1
      Serial Number (system): [REDACTED]
      Hardware UUID: [REDACTED]
      Provisioning UDID: [REDACTED]
      Activation Lock Status: Enabled
```

**Comment by jmagman** [comment]
I'm going to close this again as this is a bug in the iOS simulator, and it reproduces without Flutter if you have x64 code running on an arm Apple Silicon simulator.

If you're seeing this, you should try to identify which plugin has `arm64` in the **Excluded Architectures > Any iOS Simulator SDK** build setting
![excluded_archs](https://github.com/user-attachments/assets/73d3a5bb-d4b2-4a1f-9aaa-ccef76063c96)

And then update to a version of that plugin that supports arm64 (for example, google_maps_flutter started supporting arm simulators in 2.6.0), or if none exists, work with the plugin author.
